invenio_swh/service.py

```diff
@@ -127,9 +130,34 @@ def sync_status(self, id_, uow=None):
         # Handle swhid created
         swhid = res.get("deposit_swhid")
         if swhid and not deposit.swhid:
+            try:
+                record = record_service.record_cls.get_record(str(deposit.record_id))
+                frecord = next(iter(record.files.entries.values()))
+                root_dir = self._peak_zip_root(frecord)
+                if root_dir:
+                    swhid = (
+                        swhid.replace("path=/", f"path={root_dir}")
+                        if "path=/" in swhid
+                        else f"{swhid};path={root_dir}"
+                    )
+            except Exception:
+                # If the `path` cannot be determined, the swhid points to the root of the deposit
+                pass
             self.update_swhid(deposit.id, swhid, uow=uow)
         return self.result_item(deposit)
 
+    def _peak_zip_root(self, file):
+        """Peaks the root of the zip file and returns the directory name."""
+        path = None
+        try:
+            with file.get_stream(mode="rb") as fp:
+                if is_zipfile(fp):
+                    (top_level_path,) = Path(fp).iterdir()
+                    path = top_level_path.name
```

Comment:
minor: I was initially thinking if what we actually care about GitHub releases only (since they have the ZIP problem). But maybe the better approach is to make sure that there's only one top-level directory (and no other top-level files), and it's somewhat safe to apply?